app/controllers/foreman_wreckingball/hosts_controller.rb

```diff
@@ -27,12 +27,15 @@ def status_dashboard
       @newest_data = Host.authorized(:view_hosts, Host).joins(:vmware_facet).maximum('vmware_facets.updated_at')
       @data = statuses.map do |status|
         host_association = status.host_association
-        counter = Host.authorized(:view_hosts, Host)
+
+        # Let the database calculate status counts, then map to HostStatus::Global values
+        counter = Host.authorized(:view_hosts)
                       .joins(host_association)
-                      .includes(host_association)
-                      .map { |host| host.public_send(host_association).to_global }
-                      .group_by { |global_status| global_status }
-                      .each_with_object({}) { |(global_status, items), hash| hash[global_status] = items.size }
+                      .select('host_status.status')
+                      .group('host_status.status')
+                      .count
+                      .each_with_object({}) { |r, (k, v)| r[status.to_global(k)] = v }
```

Comment:
it returns empty hash, I think the order of params is incorrect
```suggestion
                      .each_with_object({}) { |(k, v), r| r[status.to_global(k)] = v }
```

Reply:
Hi, thanks for pointing out. :+1: